### PR TITLE
(0.3.0) Cleaner syntax for ECCOFieldTimeSeries and ECCORestoring

### LINKED ISSUE #231: `ECCORestoring` method accepts architecture and grid; validation required?
Do we want to check whether `architecture` is different from that of grid's? Does it matter?

https://github.com/CliMA/ClimaOcean.jl/blob/2e3ce57f23922dd557abe56b111eb380b39f987d/src/DataWrangling/ECCO/ECCO_restoring.jl#L309-L316

### PR DESCRIPTION
**Pull request by glwagner**
Closes #231

### CODE REVIEW

**Comment by glwagner** [comment]
It's still kind of confusing how `ECCORestoring` takes both an arch and grid to me

**Comment by navidcy** [comment]
> It's still kind of confusing how `ECCORestoring` takes both an arch and grid to me

Agree 👍🏼 
https://github.com/CliMA/ClimaOcean.jl/issues/231

**Comment by glwagner** [comment]
Ok, I will clean up `ECCORestoring` here as well

**Comment by glwagner** [comment]
Still TODO here is to clean up the constructor for `ECCORestoring`, but I plan to do this along the same lines as the one for `ECCOFieldTimeSeries`

**Review by simone-silvestri — APPROVED** [review]

File: src/DataWrangling/ECCO/ECCO_restoring.jl [hunk 1]
```diff
@@ -138,49 +135,47 @@ Keyword Arguments
...
-function ECCO_field_time_series(metadata::ECCOMetadata;	
-                                architecture = CPU(),	
-                                time_indices_in_memory = 2,	
-                                time_indexing = Cyclical(),
-                                inpainting = NearestNeighborInpainting(prod(size(metadata))),
-                                cache_inpainted_data = true,
-                                grid = nothing)	
+function ECCOFieldTimeSeries(metadata::ECCOMetadata, arch::AbstractArchitecture=CPU(); kw...)
+    download_dataset(metadata)
+    ftmp = empty_ECCO_field(first(metadata); architecture)
```
Comment (navidcy) [inline src/DataWrangling/ECCO/ECCO_restoring.jl:140]:
```suggestion
    ftmp = empty_ECCO_field(first(metadata); arch)
```

File: src/DataWrangling/ECCO/ECCO_restoring.jl [hunk 2]
```diff
@@ -248,7 +249,7 @@ Build a forcing term that restores to values stored in an ECCO field time series
 The restoring is applied as a forcing on the right hand side of the evolution equations calculated as
 
 ```math
-Fψ = μ r (ψ_ECCO - ψ)
+Fψ = r μ (ψ_ECCO - ψ)
```
Comment (navidcy) [inline src/DataWrangling/ECCO/ECCO_restoring.jl:252]:
```suggestion
Fψ = r μ (ψ_{ECCO} - ψ)
```

**Review by simone-silvestri — APPROVED** [review]

**Pull request merged**